pkg/square/square_fuzz_test.go

```diff
@@ -70,3 +70,52 @@ func FuzzSquare(f *testing.F) {
 		}
 	})
 }
+
+// FuzzSquareDeconstruct tests whether square deconstruction function can correctly deconstruct a block back from a given square.
+func FuzzSquareDeconstruct(f *testing.F) {
+	encCfg := encoding.MakeConfig(app.ModuleEncodingRegisters...)
+
+	f.Add(0, 1)
+	f.Fuzz(func(t *testing.T, normalTxCount int, pfbCount int) {
+		// skip negative values
+		if normalTxCount < 0 || pfbCount < 0 {
+			t.Skip()
+		}
+		maxBlobSize := 1000 // @TODO there might be a global constant for this
+		allTxs := GenerateOrderedRandomTxs(encCfg.TxConfig, normalTxCount, pfbCount, maxBlobSize)
```

Comment:
why not use `generateOrderedTxs`

Reply:
Well the main difference AFAICS between the two functions is that `generateOrderedTxs()` allows you to specify the blobsPerPFB as an argument while `GenerateOrderedRandomTxs()` generates a random number within the function. As I alluded to, the fuzzer should really be responsible for all randomness hence why I prefer `generateOrderedTxs()`

Reply:
> the fuzzer should really be responsible for all randomness 

I am onboarded and agree with the idea of delegating randomness generation to the fuzzer, yet, it is not the case even in the `generateOrderedTxs()` function. It introduces internal randomness when generating transactions, that is, with the same inputs it produces different outputs each time. Please see the following sample:
```go
func FuzzOrderedTxsRandmoness(f *testing.F) {
	var (
		normalTxCount uint = 12
		pfbCount      uint = 91
		blobsPerPfb   uint = 2
		blobSize      uint = 812
	)
	f.Add(normalTxCount, pfbCount, blobsPerPfb, blobSize)
	f.Fuzz(func(t *testing.T, normalTxCount, pfbCount, blobsPerPfb, blobSize uint) {
		set1 := generateOrderedTxs(int(normalTxCount), int(pfbCount), int(blobsPerPfb), int(blobSize))
		set2 := generateOrderedTxs(int(normalTxCount), int(pfbCount), int(blobsPerPfb), int(blobSize))
		require.Equal(t, set1, set2) // this fails, the transactions are not the same, regardless of the proportion of normal and pfb transactions i.e., normalTxCount and pfbCount
	})

}
```
There are many reasons why they are not the same including (but not limited to) the randomness produced internally in the [`RandBlobTxs`](https://github.com/celestiaorg/celestia-app/blob/53abe8824af0988d816bb241623b59a8d61387b3/pkg/square/builder_test.go#L63) which invokes [`RandMsgPayForBlobsWithSigner`](https://github.com/celestiaorg/celestia-app/blob/cb2d7ea4648d97ff32985cdd401a56ecda61de32/test/util/blobfactory/payforblob_factory.go#L238) which generates random blobs and namespaces.  

Considering the above observation (the internal randomness in the transaction generation), I incorporated additional randomness into the transaction generation process to provide further confidence in the absence of bugs in our code. This decision was based on the implicit assumption that our code is bug-free, but we wanted to validate this assumption. By generating more randomized data and testing our implementation against it, the aim is to obtain an additional guarantee of bug-free behavior. If our assumption holds true, the fuzz testing should not uncover any failing inputs. However, if we do encounter a failing input, we would inevitably need to rely on the logged error messages for debugging, considering the inherent randomness present in almost all of our test fixtures. 

Having said that, I am fine with merging the two tests. However, I want to highlight that the point you mentioned about there being "no outside randomness" would not be satisfied in either scenario, as far as I can tell. We may need actually do further refactoring to achieve this.  Please let me know your thoughts.